shell/initialize/client.js

```diff
@@ -543,108 +538,6 @@ function nuxtReady(_app) {
   });
 }
 
-const noopData = () => {
-  return {};
-};
-const noopFetch = () => {};
-
-// Special hot reload with asyncData(context)
-function getNuxtChildComponents($parent, $components = []) {
-  $parent.$children.forEach(($child) => {
-    if ($child.$vnode && $child.$vnode.data.nuxtChild && !$components.find((c) => (c.$options.__file === $child.$options.__file))) {
-      $components.push($child);
-    }
-    if ($child.$children && $child.$children.length) {
-      getNuxtChildComponents($child, $components);
-    }
-  });
-
-  return $components;
-}
-
-function hotReloadAPI(_app) {
-  if (!module.hot) {
-    return;
-  }
-
-  const $components = getNuxtChildComponents(_app.$nuxt, []);
-
-  $components.forEach(addHotReload.bind(_app));
-}
```

Comment:
I got to this when removing `getNuxtChildComponents`. I thought this might break hot reloading with asyncData components but it looks like hot reloading is already broken without these changes. We should just move the existing asyncData components to fetch.